R/type-tibble.R

```diff
@@ -20,3 +20,30 @@ vec_ptype2.tbl_df.data.frame <- function(x, y, ..., x_arg = "x", y_arg = "y") {
 vec_ptype2.data.frame.tbl_df <- function(x, y, ..., x_arg = "x", y_arg = "y") {
   df_as_tibble(.Call(vctrs_type2_df_df, x, y, x_arg, y_arg))
 }
+
+
+tbl_ptype2.tbl_df <- function(x, y, ..., x_arg = "x", y_arg = "y") {
+  if (inherits_only(x, c("tbl_df", "tbl", "data.frame"))) {
+    UseMethod("tbl_ptype2.tbl_df", y)
+  } else {
+    stop_incompatible_type(x, y, x_arg = x_arg, y_arg = y_arg)
+  }
+}
+tbl_ptype2.tbl_df.default <- function(x, y, ..., x_arg = "x", y_arg = "y") {
+  stop_incompatible_type(x, y, x_arg = x_arg, y_arg = y_arg)
+}
+
+tbl_ptype2.tbl_df.data.frame <- function(x, y, ..., x_arg = "x", y_arg = "y") {
+  if (inherits_only(y, "data.frame")) {
+    tibble::tibble()
+  } else {
+    stop_incompatible_type(x, y, x_arg = x_arg, y_arg = y_arg)
+  }
+}
+tbl_ptype2.data.frame.tbl_df <- function(x, y, ..., x_arg = "x", y_arg = "y") {
+  if (inherits_only(y, c("tbl_df", "tbl", "data.frame"))) {
```

Comment:
Should this be `inherits_exactly()`?

Reply:
This is from an old version of rlang, 0.2.0 I believe. Do you think it's worth to rename?